src/frontend/js/components/AddressesManagement/useAddressesManagement.tsx

```diff
@@ -0,0 +1,135 @@
+import { messages } from 'components/AddressesManagement/index';
+import { AddressesMutateOptions, useAddresses } from 'hooks/useAddresses';
+import { useState } from 'react';
+import { useIntl } from 'react-intl';
+import { MutateOptions } from 'react-query';
+import * as Joanie from 'types/Joanie';
+import { Address } from 'types/Joanie';
+import { Maybe } from 'types/utils';
+import { HttpError } from 'utils/errors/HttpError';
+import { confirm } from 'utils/indirection/window';
+
+/**
+ * High-level helper to handle various `Address` CRUD with loading and error built-in.
+ */
+export function useAddressesManagement() {
+  const intl = useIntl();
+  const addresses = useAddresses();
+  const [error, setError] = useState<Maybe<string>>();
+  const [load, setLoad] = useState(false);
+
+  /**
+   * Update the provided address to promote it as main
+   *
+   * @param {Joanie.Address} address
+   */
+  const promoteAddress = (address: Joanie.Address) => {
+    if (!address.is_main) {
```

Comment:
```suggestion
    setError(undefined);
    if (!address.is_main) {
```
In case address is main, shouldn't we set the setError to undefined here?

Reply:
ditto